tests/tests.sh

```diff
@@ -158,15 +165,15 @@ function compare_to_result()
 	local sav="$2"
 	if [ ! -r "$sav" ]; then return 0; fi
 	test "`cat -- "$sav"`" = "UNDEFINED" && return 1
-	diff -u -- "$sav" "$outdir/$pattern.res" > "$outdir/$pattern.diff.sav.log"
+	diff -u --strip-trailing-cr -- "$sav" "$outdir/$pattern.res" > "$outdir/$pattern.diff.sav.log"
```

Comment:
There was a few wrong assumptions here, so just to clarify:

* The toolchain inserts the CRs, so all the tests output CRs. Most of the time it's useful, other times it is not.
* This is only about *trailing* CRs in *sav/res* files, not about *CRs in any output*. For instance, the binary part of the `binary` tests don't use the sav/res files so they are not affected.